ubi_manifest/worker/tasks/auditing.py

```diff
@@ -0,0 +1,355 @@
+                    out_repo.id,
+                )
+                continue
+            if repo_type == "bin_repo":
+                _LOG.info(
+                    "Processing and auditing UBI repo '%s' with modular content...",
+                    out_repo.id,
+                )
+                _LOG.warning(
+                    "Only auditing of non modular content has been implemented.\n"
+                )
+            else:
+                _LOG.info("Processing and auditing UBI repo '%s'...\n", out_repo.id)
+            self.nonmodular_auditor.out_repo_id = out_repo.id
+            self._process_and_audit_type(out_repo, in_repos, repo_type)
+
+    def _process_and_audit_type(
+        self,
+        out_repo: YumRepository,
+        in_repos: List[YumRepository],
+        repo_type: str,
+    ) -> None:
+        """
+        Processes and audits the repository content.
+
+        Fetches content from the output and input repositories, sets the
+        whitelist and blacklist, and performs validation and checks.
+        """
+        self._set_whitelist_blacklist(out_repo, in_repos, repo_type)
+        self._fetch_out_repo_content(out_repo)
+        self._fetch_in_repos_contents(in_repos)
+        self.nonmodular_auditor.validate_versions()
+        self.nonmodular_auditor.check_content_rules()
+
+    def _fetch_out_repo_content(self, out_repo: YumRepository) -> None:
+        """
+        Fetches the content of the output repository.
+
+        Updates the nonmodular RPM units in the output repository content
+        and the auditor.
+        """
+        all_rpm_units: set[UbiUnit] = search_rpms(
+            [Criteria.true()], [out_repo], BATCH_SIZE_RPM
+        ).result()
+        non_modular_rpms = get_n_latest_from_content(
+            all_rpm_units, modular_rpms=self.all_modular_filenames
+        )
+        self.out_repo_content.nonmodular_rpm_units = set(non_modular_rpms)
+        # arranging units by name and architecture
+        self.nonmodular_auditor.arranged_out_units = {}
+        for unit in self.out_repo_content.nonmodular_rpm_units:
+            self.nonmodular_auditor.arranged_out_units[(unit.name, unit.arch)] = unit
+
+    def _fetch_in_repos_contents(self, in_repos: List[YumRepository]) -> None:
+        """
+        Fetches the contents of the input repositories.
+
+        Updates the nonmodular RPM units in the input repositories content
+        and the auditor.
+        """
+        fields = ["name", "arch"]
+        names_arch = {
+            (unit.name, unit.arch)
+            for unit in self.out_repo_content.nonmodular_rpm_units
+        }
+        default_arch = self.out_repo_bundle["bin_repo"].arch
+        _LOG.info("default arch: %s\n", default_arch)
+        for pkg in self.nonmodular_auditor.whitelist:  # type: ignore
+            if not any(name == pkg for name, _ in names_arch):
+                names_arch.add((pkg, default_arch))
```

Comment:
As I mentioned above, this may lead to inaccurate results.
For any pkg from whitelist you shouldn't use `default_arch`, just query by `name` will be good enough.